pygsti/report/workspaceplots.py

```diff
@@ -662,42 +697,82 @@ def _circuit_color_scatterplot(circuit_structure, sub_mxs, colormap,
+        elif isinstance(g, _CircuitList) or (isinstance(g, list) and all([isinstance(el, _CircuitList) for el in g])):
+            hover_info = _create_hover_info_fn_circuit_list(circuit_structure, sum_up, addl_hover_submxs)
+        
     xs = []; ys = []; texts = []
     gstrs = set()  # to eliminate duplicate strings
-    for ix, x in enumerate(g.used_xs):
-        for iy, y in enumerate(g.used_ys):
-            plaq = g.plaquette(x, y, empty_if_missing=True)
-            if sum_up:
-                if plaq.base not in gstrs:
-                    tot = sum([sub_mxs[iy][ix][iiy][iix] for iiy, iix, _ in plaq])
-                    xs.append(len(plaq.base))  # x-coord is len of *base* string
-                    ys.append(tot)
-                    gstrs.add(plaq.base)
-                    if hover_info:
-                        if callable(hover_info):
-                            texts.append(hover_info(tot, iy, ix))
-                        else:
-                            texts.append(str(tot))
+    
+    if isinstance(g, _PlaquetteGridCircuitStructure):
+        for ix, x in enumerate(g.used_xs):
+            for iy, y in enumerate(g.used_ys):
+                plaq = g.plaquette(x, y, empty_if_missing=True)
+                if sum_up:
+                    if plaq.base not in gstrs:
+                        tot = sum([sub_mxs[iy][ix][iiy][iix] for iiy, iix, _ in plaq])
+                        xs.append(len(plaq.base))  # x-coord is len of *base* string
+                        ys.append(tot)
+                        gstrs.add(plaq.base)
+                        if hover_info:
+                            if callable(hover_info):
+                                texts.append(hover_info(tot, iy, ix))
+                            else:
+                                texts.append(str(tot))
+                else:
+                    for iiy, iix, opstr in plaq:
+                        if opstr in gstrs: continue  # skip duplicates
+                        xs.append(len(opstr))
+                        ys.append(sub_mxs[iy][ix][iiy][iix])
+                        gstrs.add(opstr)
+                        if hover_info:
+                            if callable(hover_info):
+                                texts.append(hover_info(sub_mxs[iy][ix][iiy][iix], iy, ix, iiy, iix))
+                            else:
+                                texts.append(str(sub_mxs[iy][ix][iiy][iix]))
+    elif isinstance(g, _CircuitList):
+        for i, ckt in enumerate(g):
+            if ckt in gstrs:
+                continue
             else:
-                for iiy, iix, opstr in plaq:
-                    if opstr in gstrs: continue  # skip duplicates
-                    xs.append(len(opstr))
-                    ys.append(sub_mxs[iy][ix][iiy][iix])
-                    gstrs.add(opstr)
+                if sum_up:
+                    pass
+                    #TODO: Implement sum_up behavior mirroring that above.
```

Comment:
Is this something we should implement now before it leaves mental cache? I can't imagine we want to touch this code very frequently.

I think this is something I can take a stab at also - we just need to iterate through the circuit lists, collect/sum the information, and store it as a summary_label attribute to be used in the hover_fn construction function?

Reply:
The ship has long since sailed on this leaving my mental cache, unfortunately. However, in starting the process of trying to implement this I realized why it was I didn't do so earlier. TLDR: I haven't figured out exactly what the `sum_up` option should actually do for circuit lists (or lists of circuit lists). For plaquettes what this does is it sums over the constituents of a single plaquette and then adds a scatter point to the figure with a y value given by the sum over the plaquette, and an x-value corresponding to the length of the base circuit (germ raised to some power). So the meaning of this action is fairly clear. I'm not sure how we'd meaningfully port this to CircuitList or lists of CircuitList objects though, since I don't know what x would be. If we had some additional structural guarantees/metadata we could do so, but presently we don't. (Incidentally, the thought did cross my mind last year that it may be useful to add a new circuit structure as a wrapper around CircuitList objects that provided this sort of additional metadata, but obviously I never did so).

Reply:
Fair enough, I also didn't think it through that far. Let's leave it as is in that case, and we can revisit this if we ever do something with a container of CircuitList objects (which is an interesting idea, but agreed that it's a relatively low priority).